src/linker/configuration.py

```diff
@@ -42,20 +44,20 @@ class Config:
 
     def __init__(
         self,
-        pipeline_specification: Path,
-        input_data: Path,
-        computing_environment: Optional[Path],
+        pipeline_specification: Union[str, Path],
```

Comment:
we really don't rely on Path very much here, so I made it compatible with str or Path inputs (wrapping `Path()` as needed)

Reply:
Is it even possible to be a Path at this point? This only gets instantiated from cli.py, right?

Reply:
You're right; we determine exactly what the input data type is so we can choose to pass in the "correct" one and just type hint for that if we want.
IME though in development it's not always easy to track whether you need to be passing in a string path or Path object and often need to convert back and forth, so highlighting that its compatible with either can be useful